test/mako/config/benchmark.go

```diff
@@ -25,39 +25,44 @@ import (
 
 	"github.com/golang/protobuf/proto"
 	mpb "github.com/google/mako/spec/proto/mako_go_proto"
+	"github.com/pkg/errors"
 )
 
 const koDataPathEnvName = "KO_DATA_PATH"
+const configMako = "/etc/config-mako"
 
 // MustGetBenchmark wraps getBenchmark in log.Fatalf
-func MustGetBenchmark() (*string, *string) {
-	benchmarkKey, benchmarkName, err := getBenchmark()
+func MustGetBenchmark() *mpb.BenchmarkInfo {
+	bench, err := getBenchmark()
 	if err != nil {
-		log.Fatalf("unable to determine benchmark_key: %v", err)
+		log.Fatalf("unable to determine benchmark info: %v", err)
 	}
-	return benchmarkKey, benchmarkName
+	return bench
 }
 
 // getBenchmark fetches the appropriate benchmark_key for this configured environment.
-func getBenchmark() (*string, *string, error) {
+func getBenchmark() (*mpb.BenchmarkInfo, error) {
 	// Figure out what environment we're running in from the Mako configmap.
 	env, err := getEnvironment()
 	if err != nil {
-		return nil, nil, err
+		return nil, err
 	}
 	// Read the Mako config file for this environment.
 	data, err := readFileFromKoData(env + ".config")
 	if err != nil {
-		return nil, nil, err
+		data, err = ioutil.ReadFile(filepath.Join(configMako, env+".config"))
```

Comment:
If we support adding the benchmark config in the configmap, then `SetupWithBenchmarkConfig` in https://github.com/knative/pkg/blob/d9e3e244a6c044379b71d90b7440daad5b7910f3/test/mako/sidecar.go#L183 will not be needed.
@grantr do you think we can change to this implementation?

Reply:
Yes, I think we can change to this implementation. The current solution isn't sufficient to enable different configs in prod/dev environments, and benchmark config in ConfigMaps fixes that.

We'll need to add "also update the ConfigMaps" to the instructions for updating benchmark config files.

Reply:
And yes I think we can remove `SetupWithBenchmarkConfig` now.